frontend/src/modules/editor/components/KeyboardShortcuts.js

```diff
@@ -0,0 +1,154 @@
+/* @flow */
+
+import * as React from 'react';
+import onClickOutside from 'react-onclickoutside';
+import { Localized } from 'fluent-react';
+
+import './KeyboardShortcuts.css';
+
+
+type Props = null;
+
+
+type State = {|
+    visible: boolean,
+|};
+
+
+/*
+ * Shows a list of keyboard shortcuts.
+ */
+export class EditorSettingsBase extends React.Component<Props, State> {
+    constructor() {
+        super();
+        this.state = {
+            visible: false,
+        };
+    }
+
+    toggleVisibility = () => {
+        this.setState({
+            visible: !this.state.visible,
+        });
+    }
+
+    // This method is called by the Higher-Order Component `onClickOutside`
+    // when a user clicks outside the keyboard shortcut overlay.
+    handleClickOutside = () => {
+        this.setState({
+            visible: false,
+        });
+    }
+
+    render() {
+        return <div className="keyboard-shortcuts">
+            <Localized
+                id="keyboard-shortcuts-button"
+                attrs={{ title: true }}
+            >
+                <div
+                    className="selector far fa-keyboard"
+                    title="Keyboard Shortcuts"
+                    onClick={ this.toggleVisibility }
+                />
+            </Localized>
+
+            { !this.state.visible ? null :
+            <div
+                className="overlay"
+                onClick={ this.toggleVisibility }
+            >
+                <Localized id="keyboard-shortcuts-overlay-title">
+                    <h2>Keyboard Shortcuts</h2>
+                </Localized>
+                <table>
+                    <tbody>
+                        <tr>
+                            <Localized id="keyboard-shortcuts-save-translation">
```

Comment:
So far I've added the module name to those IDs, so in this case I would have called it `editor-keyboard-shortcuts-save-translation`. The idea was to avoid name conflicts as much as possible. Now I'm not really sure if there's value in doing that. I also wonder how much we should share strings between components (for example, `History`, `Machinery` and `OtherLocales` all use the same `No translations` string, and so I used the same ID for all three). I'd like to know what you think about all that.

I'm mostly jumping on the opportunity to start a discussion about IDs, but that's definitely not a blocking point for this PR.

Reply:
Let's stick to the tradition you started for now. I'll update the IDs.

We'll probably see later on, when we'll have all strings "extracted" - it might even make sense to split them into multiple files.